### ISSUE TITLE
Import error with Firebase 3 (firebase-bower) in node

### ISSUE THREAD

**Issue by dinoboff** [jspm/jspm-cli#2036]
When installing "firebase@3", it works fine in the browser, but fails in node.

I install it with the following overwrite:
```
jspm install firebase@3 -o '{main: "firebase", format: "global", meta: {"firebase.js": {exports: "firebase", format: "global"}}}'
```

In node with something like (for testing with mocha):
```js
const jspm = require('jspm');
const loader = new jspm.Loader();

loader.import('firebase').then(
  m => console.log(m)
).catch(
  e => console.error(e)
);
```

I get:

    { [Error: (SystemJS) firebase is not defined
      ReferenceError: firebase is not defined
          at path/to/jspm_packages/github/firebase/firebase-bower@3.3.0/firebase.js:28:1
          at Object.exports.runInThisContext (vm.js:54:17)
      Evaluating path/to/jspm_packages/github/firebase/firebase-bower@3.3.0/firebase.js
      Error loading path/to/jspm_packages/github/firebase/firebase-bower@3.3.0/firebase.js] originalErr: [ReferenceError: firebase is not defined] }

firebase.js:28:1
```
  [...]
  "undefined" !== typeof window && (window.firebase = Ja());
})();
firebase.SDK_VERSION = "3.3.0";
(function() {
  [...]
```

**Comment by guybedford** [comment]
Is this with jspm 0.16 or jspm 0.17? I'd suggest trying jspm 0.17 as it has better support for switching between Node and the browser.

**Comment by dinoboff** [comment]
it's with 0.17.0-beta.25

**Comment by guybedford** [comment]
Try `jspm install npm:firebase` with the following override:

```json
{
  "main": "firebase-node.js",
  "format": "cjs",
  "meta": {
    "*": {
      "globals": {
        "process": "process"
      }
    },
    "*.json": {
      "format": "json"
    },
    "auth-node/credential.js": {
      "globals": {
        "Buffer": "buffer/global"
      }
    },
    "auth-node/token-generator.js": {
      "globals": {
        "Buffer": "buffer/global"
      }
    },
    "database-node.js": {
      "deps": ["./app-node.js", "_stream_writable", "_stream_duplex", "faye-websocket", "request"],
      "globals": {
        "Buffer": "buffer/global"
      }
    }
  },
  "map": {
    "_stream_writable": "@node/_stream_writable",
    "_stream_duplex": "@node/_stream_duplex",
    "./auth-node": "./auth-node/index.js",
    "./firebase-node.js": {
      "browser": "./firebase-browser.js"
    }
  }
}
```

**Comment by dinoboff** [comment]
I really only need import of firebase to works in my mocha tests; except for access to some constant (`firebase.database.ServerValue.TIMESTAMP`), firebase will be mocked.

I would prefer to keep using the bower version for development and self-executable bundles.

What I can do is map firebase to a mock or to node's firebase:
```
jspm install firebase@3 -o '{main: "firebase", format: "global", meta: {"firebase.js": {exports: "firebase", format: "global"}}}'
npm install firebase@3
```

```js
const jspm = require('jspm');
const loader = new jspm.Loader();

loader.config({map: {firebase: '@node/firebase'}});

loader.import('firebase').then(
  m => console.log(m)
).catch(
  e => console.error(e)
);
```

**Comment by dinoboff** [comment]
Anyway I can save "{map: {firebase: '@node/firebase'}" in jspm.config.js?

I tries to had it to nodeConfig, but is seems to be used by the build command.

**Comment by dinoboff** [comment]
It has been solved win firebase 3.4.0.
